src/ControlSystem/Controller.hpp

```diff
@@ -63,23 +63,40 @@ class Controller {
   Controller& operator=(const Controller&) = default;
   ~Controller() = default;
 
-  DataVector operator()(
-      const DataVector& timescales,
-      const std::array<DataVector, DerivOrder + 1>& q_and_derivs,
-      double q_time_offset, double deriv_time_offset) const;
+  DataVector operator()(const double time, const DataVector& timescales,
+                        const std::array<DataVector, DerivOrder>& q_and_derivs,
+                        double q_time_offset, double deriv_time_offset);
 
-  /// Takes the current minimum of all timescales and uses that to set the time
-  /// between updates
+  /// Takes the current minimum of all timescales for a specific control system
+  /// and uses that to set the time between updates
   void assign_time_between_updates(const double current_min_timescale) {
     time_between_updates_ = update_fraction_ * current_min_timescale;
   }
 
+  /// Returns `true` if the controller is ready to calculate the control signal
+  bool is_ready(const double time) {
+    return time >= last_update_time_ + time_between_updates_;
+  }
+
+  /// Calculates the new expiration time as the old expiration time plus the
+  /// time between updates
+  double next_expiration_time(const double current_expiration_time) {
+    return current_expiration_time + time_between_updates_;
+  }
```

Comment:
No it isn't a problem. In a future PR there will be a `bool` defined as
```
const bool need_to_update_now = current_expiration_time <= time + current_time_between_measurements;
```
which will catch if the update time and expiration time have drifted too much and force an update before an error occurs.

In my testing, omitting this bool caused exactly the error you are describing. This is not added in the averager or the controller because both require this bool to determine if they are ready, so it is external.

If you think it's easier to understand, I can add it as an argument to the `is_ready` function for both the averager and controller.

Reply:
If it is handled elsewhere that is fine.